### DISCUSSION TITLE
Taking a little break

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#774] [Announcements]
Hello.

I've been working on the integration for a little while now, with what I feel includes regular releases. I never thought this integration would be as popular as it is. However, with this popularity more feature requests, discussions  and the occasional bugs are being raised every day, in addition to surprise changes in the OE API and HA. The result is that I'm now feeling quite burnt out maintaining this integration alongside with my day job. If I'm honest, I've not been very good at putting up boundaries.

My plan is not to stop developing the integration, but slow down. Therefore, to try and regenerate I will not be responding to issues or discussions for the month of March, but should hopefully be back in April at a reduced pace.

Thanks,
Dave :)

**Comment by alienatedsec** [comment]
Hi @BottlecapDave

Thanks for your efforts so far - I can fully relate as I have done similar; however, I did realise contributing to some of my projects impacts my family time and more importantly, other areas where I did put myself last.

Warm regards

**Comment by SteveDrakey** [comment]
Thanks for the efforts so far, have you thought of pairing up with megakid ? both integrations are pretty good no they are both AWSOME !!! having different takes on how to surface them to HA.

I also think Andy who makes Octo aid would create some magically things if he helped out with anything related to Octopus and HA.

Thanks again !!

**Reply by BottlecapDave** [reply]
The initial intelligent integration was based on an interest from megakid to combine the two as he was about to have a busier personal life and couldn't dedicate time to the integration. The initial release was based heavily on the work he'd done, with his approval. Ive even updated the docs with how to migrate, but not sure if there's something else missing.

I'm always open for people to contribute, but I also don't want people to spend large amounts of effort to only find out it's not a right fit for the integration. This is why I mark issues as such in the backlog once I feel I've specd them out to a sufficient level :)

**Comment by SteveDrakey** [comment]
Thanks for the efforts so far, have you thought of pairing up with megakid ? both integrations are pretty good no they are both AWSOME !!! having differnt takes on how to surface them to HA.

I also think Andy who makes Octo aid would create some magically things if he helped out with anything related to Octopus and HA.

Thanks again !!

**Comment by dan-r** [comment]
Thanks for your work on this Dave. The amount of functionality is mindblowing and this integration makes up for a fair chunk of my home automation, its clear that you've put **a lot** of time into it and are well due a break!

I'm happy to have a go at resolving some bugs and helping with some troubleshooting going forwards, and think its something the developers among us could do more of - with a user base this size its not sustainable to leave it all on you! (I know its a fine balance between helping out and too many cooks in the kitchen though 🙂)

**Reply by BottlecapDave** [reply]
Thanks for the interest. Debugging and fixing issues for tariffs I'm not apart of or can't easily test is always a big help :)

**Comment by RobinXe** [comment]
Thanks for all you've given us so far Dave; you've certainly gone above and beyond, and produced a fantastic product!

Enjoy a well-earned break from it. It's a healthy decision!

**Comment by pacemaker82** [comment]
Thanks Dave for your hard work - its a great integration and very appreciated! Hopefully the community can help with the bug fixing to take the edge off

**Comment by iainfogg** [comment]
Hi @BottlecapDave thanks so much for all your work on this - as others have said, you've developed something amazing, but that's not more important than your wellbeing. I really hope this time out helps you to get yourself to a better place, and when you come back, I'd certainly encourage you to think about how much you call the integration feature-complete, vs. how much you want to see it as a continually evolving thing - just because it could do something doesn't mean that it should do something, with all the dev, testing, bug fixing etc. that that entails.

During the time you're off, I wondered if there's anyone who would be able to at least coordinate, merge and tag / release on the repo if some breaking change suddenly appears in the Octopus API - I'd be happy to do that, maybe with one or two others, if that could do that aspect if that's helpful. That way, during your absence in March, we can at least make sure it stays functional. What do you think? Then it's open for anyone to do the urgent bug fixing dev work, but we know we can do a merge and release.

And to be clear, I'm talking about urgent bug fixes only - not features etc.

Reply and tag me if that would help.

PS I'm involved on the Predbat repo currently, so I'm definitely keen to see your integration thriving.

**Reply by BottlecapDave** [reply]
Is software ever complete :)?

I'm always up for the integration to evolve and add new features. If a feature request comes in that i feel ill use or a lot of users would take advantage of, I'm ok implementing it (to a degree). For the other feature requests, if it can be accomplished with existing HA technology (e.g. a template sensor) then I'll either reject it and suggest the alternative or it will go to the bottom of the list. If it can't then I try and tackle it over time (see multi account support) or I'll spec it out and mark it as such in the issues list. This latter option prob needs to be advertised more. When it's marked as such, it's not to say I won't get to it, but it's not a priority.

If something breaks that cant wait a month or has a workaround, I'm sure I'll see it in my own instance and will probably jump on it :) but I'm hopefull that neither HA or octopus API will change significantly

**Comment by psfshr** [comment]
Hi @BottlecapDave,
Though I've only just got into HA, this was the first integration I added, so all the work you've done to get it to the position it's in is appreciated :-) Hope you manage to get some recovery in!

**Comment by seanmaskey** [comment]
Hi - Im glad to hear you're taking a well earned break - "annual leave" is generally a good thing.

An observation - having just got an Octopus Mini in the new roll out.

I connected it & followed the app integration & got the OK & green light. App said data would flow shortly & I  tickied the, I have a mini box. Bad move, as almost all the automations based on the agile tariff turned off.  They came back on when I unticked the box 24 hrs later. I had assumed the mini entities were additions & there would be no change to existing setup. Clearly not the case & that may be obvious to those who understand the details of HA & the integration, but it caught me out. - at least I assume this is why I'm getting unexpected behaviour with the mini.

**Reply by BottlecapDave** [reply]
It was much needed thanks :)

Ticking the home mini should not have any effect on existing entities and is indeed additive. I think I may have had one other report of this happening, but there is no evident reason why it should (other than a potentially dodgy install by HACS). It definately shouldn't have turned off any automations. If you are still experiencing this issue, please raise an issue with any HA related logs attached.

**Reply by seanmaskey** [reply]
HI
according to Octopus, the smart meter connection hub is disconnected - I made the foolish error of assuming the green light on the mini meant that it is working correctly - it doesn't, just indicates its connected to the wifi seemingly. I set HA integration back to no mini & all worked fine again. still waiting for OP to tell me problem is fixed before trying mini again.

not entirely convinced by OP explanation, as its clearly sending the data out as I get daily readings 24 hrs later in their app; unless that is via a different route.

Thanks for the integration, it's great. I have an improvement idea for heat pump operation I'll post separately.

Seán

**Reply by BottlecapDave** [reply]
Ok no problem. OE will still be getting readings from your smart meter outside of your Mini, which are usually around 24 hours behind. which is what you're probably seeing in the app. The Home Mini just makes this data available in a more real time setting.

There has already been a [suggestion](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/811) around heat pumps in case this is what you were thinking.

**Comment by wickzter** [comment]
@BottlecapDave Just wanted to say thank you for your time and effort to create this integration and give it to the community.  I have just made a one off sponsorship, please have a beer (or beverage of choice) on me during your time off.
